File: perl/example1/example1.pl
#!/usr/local/bin/perl

use strict;
use IO::Socket::SSL;
use JSON::PP;
use Compress::Zlib;
use Data::Dumper;

my $username = 'XXXXXXXXXX';
my $apikey = 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX';
my $compression = 0;

# Open the TLS socket connection to FlightAware.
my $sock = IO::Socket::SSL->new('firehose-test.flightaware.com:1501') or die $!;
print "Connected!\n";

# Send the initiation command to the uncompressed socket.
my $initcmd = "live version 7.0 user $username password $apikey events \"flightplan position\"";
if ($compression) {
    $initcmd .= " compression compress";
}
binmode $sock;
print $sock "$initcmd\n";

# Activate compression, if requested.
my $zsock;
if ($compression) {
    $zsock = inflateInit()
    or die "Could not initiate inflate\n";
} else {
    $zsock = $sock;
}

# Main loop, reading lines of JSON from the server.
my $i = 1;
my $buffer = "";
while (1) {
    #print "LINE $i\n";
    #print "LINE $i: ", $line, "\n";

    my $line = "";
    my $data_available = 1;

    if ($compression) {
        if (index($buffer, "\n") == -1) {
            $data_available = read($sock, $line, 8192);
            (my $output, my $status) = $zsock->inflate($line);
            $line = $output;

            $buffer = $buffer . $line;
        }

        my $rawline = "";
        my $newline_index = index($buffer, "\n");

        if ($newline_index != -1) {
            $rawline = substr($buffer, 0, $newline_index);
            $buffer = substr($buffer, $newline_index + 1);
        }

        $line = $rawline

    } else {
        $line = $sock->getline();
        if(!defined($line)) {
            $data_available = 0;
        }
    }

    if ($line ne "") {
        my $data = eval { decode_json $line };
        die "Failed to decode JSON: $line" if !defined($data) || $@;

        print "LINE $i\n" . Dumper($data);

        last if ($i++ >= 1000);
    } 

    last if (!$data_available);
    
}
close $sock;

print "All done.\n";

File: php/example1/example1.php
<?
#Always use this example if you are not using compression
#Also use this example if you are using compression, and expecting a high volume of messages

$username = 'XXXXXXXXXX';
$apikey = 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX';
$compression = 0;


// Open the TLS socket connection to FlightAware.
$fp = fsockopen("tcp://firehose.flightaware.com", 1501, $errno, $errstr, 30);
if (!$fp) {
   echo "Error connecting ($errno): $errstr\n";
   exit(1);
}
if (!stream_socket_enable_crypto($fp, true, STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT)) {
   echo "Error negotiating TLS\n";
   fclose($fp);
   exit(1);
}

echo "Connected!\n";

// Send the initiation command to the uncompressed socket.
$initcmd = "live version 7.0 user $username password $apikey events \"flightplan position\"";
if ($compression) {
    // compress, gzip, deflate
    $initcmd .= " compression deflate";
}
fwrite($fp, "$initcmd\n");

// Apply a decompression filter, if requested.
if ($compression && !stream_filter_append($fp, 'zlib.inflate', STREAM_FILTER_READ)) {
   echo "Error appending filter.\n";
   exit(1);
}

// Main loop, reading lines of JSON from the server.
$i = 1;
while (($buffer = fgets($fp)) !== false) {
    echo "LINE $i\n";
    
    $data = json_decode($buffer);
    if (json_last_error() !== JSON_ERROR_NONE) {
        echo "Error: invalid json.\n";
        echo $buffer;
        break;
    }
    print_r($data);

    if ($i++ >= 10) {
        break;
    }
}
fclose($fp);
echo "All done.\n";

?>